go-controller/pkg/crd/nodednsinfo/v1/types.go

```diff
@@ -0,0 +1,51 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +resource:path=nodednsinfo
+// +kubebuilder:resource:scope=Cluster
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="nodeDNSInfo Status",type=string,JSONPath=".status.status"
+// NodeDNSInfo describes the current DNS Resolution for each each node for egressFirewallDNS
+// entries in the cluister.
+// This is a centralized object to coordinate nodes DNS resolution. All nodes
+// update there NodeDNSInfo object and the master node accesses the ovn database when needed.
+type NodeDNSInfo struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of EgressFirewallDNS
+	Spec NodeDNSInfoSpec `json:"spec,omitempty"`
+
+	// Observed status of EgressFirewallDNS
+	// +optional
+	Status NodeDNSInfoStatus `json:"status,omitempty"`
+}
+
+type NodeDNSInfoSpec struct {
+	// a map indexed by DNS name to its corresponding dns entry
+	DNSEntries map[string]DNSEntry `json:"egressdnsEntries,omitempty"`
```

Comment:
I still believe this should be in Status. It is pushed from the outside world into the apiserver, as opposed to the other way around.